cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -270,7 +270,14 @@ public void scanApplications(Cursor cursor,
   }
 
   public long getApplicationCount() throws IOException {
-    return getApplicationSpecificationTable().count(Range.all());
+    // Get number of applications where namespace != SYSTEM (exclude system applications)
+    Collection<Field<?>> fields = ImmutableList.of(Fields.stringField(StoreDefinition.AppMetadataStore.NAMESPACE_FIELD,
+                                                                      NamespaceId.SYSTEM.getNamespace()));
+    // Return sum of ranges [empty, SYSTEM) and (SYSTEM, empty)
+    return getApplicationSpecificationTable().count(Range.create(null, Range.Bound.INCLUSIVE, fields,
+                                                                 Range.Bound.EXCLUSIVE)) +
+      getApplicationSpecificationTable().count(Range.create(fields, Range.Bound.EXCLUSIVE, null,
```

Comment:
Can the table.count method actually take a collection of ranges such that one query is generated instead of two?